File: model/item.py
from dataclasses import dataclass, field
from datetime import datetime


@dataclass
class Item:
    name: str
    description: str
    quantity: int
    id: str = ""
    created_at: str = ""

    def __post_init__(self):
        if not self.created_at:
            self.created_at = datetime.now().strftime("%Y-%m-%dT%H:%M:%S")

    def to_dict(self) -> dict:
        return {
            "id": self.id,
            "name": self.name,
            "description": self.description,
            "quantity": self.quantity,
            "created_at": self.created_at,
        }

    @staticmethod
    def from_dict(data: dict) -> "Item":
        return Item(
            id=data["id"],
            name=data["name"],
            description=data["description"],
            quantity=data["quantity"],
            created_at=data["created_at"],
        )

File: ConsoleMVC/model/Item.cpp
#include "model/Item.h"

nlohmann::json Item::toJson() const {
    return {
        {"id",          id},
        {"name",        name},
        {"description", description},
        {"quantity",    quantity},
        {"createdAt",   createdAt}
    };
}

Item Item::fromJson(const nlohmann::json& j) {
    Item item;
    item.id          = j.at("id").get<std::string>();
    item.name        = j.at("name").get<std::string>();
    item.description = j.at("description").get<std::string>();
    item.quantity    = j.at("quantity").get<int>();
    item.createdAt   = j.at("createdAt").get<std::string>();
    return item;
}
